frontend/src/components/map.js

```diff
@@ -45,139 +46,104 @@ class WtMap extends HTMLElement {
       showElevation: mapConfig.ShowElevation,
     };
 
-    this.points = JSON.parse(this.getAttribute("map-points") || "[]");
-    if (this.points.length !== 0) {
+    this.workout = JSON.parse(
+      document.getElementById(this.getAttribute("data-el")).textContent,
+    );
+    this.preferredUnits = JSON.parse(
+      document.getElementById(this.getAttribute("preferred-units-el"))
+        .textContent,
+    );
+    if (this.workout?.positions?.Data?.length !== 0) {
```

Comment:
Just noticed that this should not be true if `this.workout?.positions?.Data` is `undefined`.